test/faas-variations/faas-variations-report.test.js

```diff
@@ -0,0 +1,166 @@
+import { expect } from '@esm-bundle/chai';
+import { getBlockInfo, init, report, loadFragments, getLetterScheme } from '../../faas-variations-report/report.js';
+import { loadDocument } from '../../bulk-update/document-manager/document-manager.js';
+import BaseReporter from '../../bulk-update/reporter/reporter.js';
+
+const { pathname } = new URL('.', import.meta.url);
+
+describe('FaaS Variations Report', () => {
+  describe('getBlockInfo', () => {
+    const tests = [
+      ['Block(option1, Option2)', { blockName: 'block', options: ['option1', 'option2'], variant: 'block (option1, option2)' }],
+      ['Block(option1)', { blockName: 'block', options: ['option1'], variant: 'block (option1)' }],
+      ['block', { blockName: 'block', options: [], variant: 'block' }],
+    ];
+
+    tests.forEach(([input, expectedOutput]) => {
+      it(`converts correct block information from '${input}' to '${expectedOutput.variant}'`, () => {
+        expect(getBlockInfo(input)).to.deep.equal(expectedOutput);
+      });
+    });
+  });
+
+  describe('getVariantName', () => {
+    const tests = [[0, 'A'], [25, 'Z'], [26, 'AA'], [51, 'AZ'], [52, 'BA'], [77, 'BZ'], [78, 'CA'], [701, 'ZZ'], [702, 'AAA']];
```

Comment:
might be good to put something unexpected in here to test mixed inputs?

Reply:
Added a test for unexpected inputs